File: 2103-rings-and-rods/2103-rings-and-rods.js
/**
 * @param {string} rings
 * @return {number}
 */
var countPoints = function(rings) {
    let map = new Map(), count=0;
    for(let i=1;i<rings.length;i+=2){
        if (!map.has(rings[i])) map.set(rings[i], new Set());
        map.get(rings[i]).add(rings[i-1]);
    }
    for(let [key, values] of map)
        if (values.size===3) count++;
    return count;
};

File: 2103-rings-and-rods/2103-rings-and-rods.java
class Solution {
    public int countPoints(String rings) {
        Map<Character, Set<Character>> map1 = new HashMap<>();
        int count=0;
        for(int i=1;i<rings.length();i+=2){
            map1.putIfAbsent(rings.charAt(i), new HashSet<>());
            map1.get(rings.charAt(i)).add(rings.charAt(i-1));
        }
        for(Set<Character> set: map1.values())
            if (set.size()==3) count++;
        return count;
    }
}